src/middlewares/authentication.js

```diff
@@ -0,0 +1,74 @@
+import jwt from 'jsonwebtoken';
+import 'dotenv/config';
+
+/**
+  * @description class representing User Authentication with JWT
+  *
+  * @class VerifyUser
+  */
+class VerifyUser {
+  /**
+    * @description - This method is responsible for generating user token
+    *
+    * @static
+    * @param {object} payload - Object representing encoded data that makes up the token
+    * @param {object} time - Object represnting time taken for the token to expire
+    *
+    * @returns {object} - object representing response message
+    *
+    * @memberof VerifyUser
+    */
+  static generateToken(payload, time) {
+    const token = jwt.sign({ payload }, process.env.TOKEN_SECRET_KEY, time);
+    return token;
+  }
+
+  /**
+    * @description - This method is responsible for creating new users
+    *
+    * @static
+    * @param {object} request - Request sent to the router
+    * @param {object} response - Response sent from the controller
+    * @param {object} next - callback function to transfer to the next method
+    * @param {object} userData - Object representing decoded data that made up the token
+    * @param {object} error - Object representing JWT error
+    *
+    * @returns {object} - object representing response message
+    *
+    * @memberof VerifyUser
+    */
+  static verifyToken(request, response, next) {
+    const token = request.headers.authorization || request.body.token;
+    if (!token) {
+      return response.status(403)
+        .json({
+          status: 'Fail',
+          message: 'No token supplied',
```

Comment:
Hello Marcus, you have done a great job! I like how you have been able to capture the right status codes for every response, but it would really be nice if you could make the error messages more descriptive. 'No token supplied' could have been 'Unauthorized Access! You are not allowed to access this route'. Also, 'Your input is not a JWT token' could have been 'Authentication failed, please login/signup'. Nice job once again, Marcus.

Reply:
Thanks for feedback @frostyblok 
The errors are separated to capture different cases. The first, "No token supplied", is meant for the case when token is empty or undefined. The second "Your input is not a JWT token" is meant for the case when a few characters are added or removed from the token. The third, "error", is meant for general cases that I can't specifically point out (cases like expired token, user supplying "abcd" as a token, etc.). This might return "malformed JWT".
However, when the APIs are consumed, the errors can be consolidated to one, say "Unauthorised Access".